crates/miden-objects/src/address/routing_parameters.rs

```diff
@@ -98,31 +120,36 @@ impl RoutingParameters {
         self.interface
     }
 
+    /// Returns the public encryption key.
+    pub fn encryption_key(&self) -> Option<SealingKey> {
+        self.encryption_key.clone()
```

Comment:
Nit: Why not return `Option<&SealingKey>` to give callers the choice when to clone? The `seal_bytes` or `unseal_bytes` methods, which are probably the most commonly used ones, only require `&SealingKey` and so cloning seems unnecessary.